src/Microsoft.AspNet.Mvc.Core/FileResult.cs

```diff
@@ -88,7 +88,7 @@ public override Task ExecuteResultAsync(ActionContext context)
                 context.HttpContext.Response.Headers[HeaderNames.ContentDisposition] = contentDisposition.ToString();
             }
             
-            logger.FileResultExecuting(FileDownloadName);
+            logger?.FileResultExecuting(FileDownloadName);
```

Comment:
Well it's up to you, my thoughts were if you allow the client to bind up an ILoggerFactory then this could happen. We could always extract the checks into some generic util class that can return either the ILogger<T> or throw an obvious error message about an ILoggerFactory returning a null logger.